public/scripts/d3/d3.js

```diff
@@ -1 +1,169 @@
-//  Add d3 functions here
+d3.json(echo.setup.jsonUrl, (err, data) => {
+  const startingCx = 160;
+  const startingCy = 120;
+
+  data.sort((a, b) => d3.descending(a.avgRating, b.avgRating));
+
+  var sortedByTag = echo.helpers.binByKey('tag', data);
+  var tagPositions = sortedByTag.map((tagObj, i) =>
+  ({ tag: tagObj.tag, cx: startingCx, cy: startingCy + 200 * i, nodes: tagObj.nodes }));
+
+  var nodeTagsArrayToLinksAndNodes = function(nodes) {
```

Comment:
okay, nodes is not an array / map of nodes, its a more complex structure, it is sortedByTags, with positions, you could call this, `tagNodesWithChildren`